mir_eval/display.py

```diff
@@ -0,0 +1,672 @@
+        pitches of notes (in Hz).
+
+    midi : np.ndarray, shape=(n,), optional
+        pitches of notes (in MIDI numbers).
+
+        At least one of `pitches` or `midi` must be provided.
+
+    kwargs :
+        Additional keyword arguments to `labeled_intervals`.
+
+    Returns
+    -------
+    ax :
+        Handle to the plotting axis
+    '''
+
+    if midi is None:
+        midi = hz_to_midi(pitches)
+
+    scale = np.arange(128)
+    ax = labeled_intervals(intervals, np.round(midi).astype(int),
+                           label_set=scale,
+                           tick=False, **kwargs)
+
+    # Minor tick at each semitone
+    ax.yaxis.set_minor_locator(MultipleLocator(1))
+
+    ax.axis('auto')
+    return ax
+
+
+def separation(sources, fs=22050, labels=None, ax=None, **kwargs):
+    '''Source-separation visualization
+
+    Parameters
+    ----------
+    sources : list of np.ndarray
+        A list of waveform buffers corresponding to each source
+
+    fs : number > 0
+        The sampling rate
+
+    labels : list of strings
+        An optional list of descriptors corresponding to each source
+
+    ax : matplotlib.pyplot.axes
+        An axis handle on which to draw the intervals.
+        If none is provided, a new set of axes is created.
+
+    kwargs :
+        Additional keyword arguments to `scipy.signal.spectrogram`
+
+    Returns
+    -------
+    ax
+        The axis handle for this plot
+    '''
+    if ax is None:
+        ax = plt.gca()
+
+    if labels is None:
+        labels = ['Source {:d}'.format(_) for _ in range(len(sources))]
+
+    specs = []
+    cumspec = None
+    kwargs.setdefault('scaling', 'spectrum')
+
+    for i, src in enumerate(sources):
+        freqs, times, spec = spectrogram(src, fs=fs, **kwargs)
+        specs.append(spec)
+        if cumspec is None:
+            cumspec = spec.copy()
+        else:
+            cumspec += spec
+
+    ref_max = cumspec.max()
+    ref_min = ref_max * 1e-6
+
+    legend_entries = []
+    for i, spec in enumerate(specs):
+
+        color = next(ax._get_lines.prop_cycler)['color']
+        cmap = LinearSegmentedColormap.from_list(labels[i],
+                                                 [(1.0, 1.0, 1.0, 0.0),
+                                                  color])
+
+        ax.pcolormesh(times, freqs, spec,
+                      cmap=cmap,
+                      norm=LogNorm(vmin=ref_min, vmax=ref_max),
+                      shading='gouraud',
+                      label=labels[i])
+
+        legend_entries.append(Patch(color=color, label=labels[i]))
+
+    handles, legend_labels = ax.get_legend_handles_labels()
+    handles.extend(legend_entries)
+    legend_labels.extend(labels)
+
+    ax.legend(handles=handles, labels=legend_labels)
+
+    __expand_limits(ax, [times.min(), times.max()], which='x')
+    __expand_limits(ax, [freqs.min(), freqs.max()], which='y')
+    return ax
+
+
+def __ticker_midi_note(x, pos):
```

Comment:
So, the main functionality here seems useful in its own right, so I would be happy to have it be non-private and factored out into a simply callable function which gave a pretty string for a given MIDI note number/float.  It'd be a bit misplaced, but useful, I think.